src/rateEngine/RateElement.ts

```diff
@@ -9,29 +9,85 @@ import LoadProfile from './LoadProfile';
 import ValidatorFactory from './ValidatorFactory';
 import { RateComponentInterface } from './RateComponent';
 import { Error } from './validators/_Validator';
+import SurchargeAsPercent from './billingDeterminants/SurchargeAsPercent';
 
 export interface RateElementInterface {
+  id?: string;
   rateElementType: RateElementType;
-  rateComponents: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
+  rateComponents?: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
   name: string;
+  billingCategory?: BillingCategory;
+  rateElementIds?: Array<string>;
+  percent?: number;
+  appliesToAll ?: boolean;
+}
+
+export enum RateElementClassification {
+  ENERGY = 'energy',
+  DEMAND = 'demand',
+  FIXED = 'fixed',
+  SURCHARGE = 'surcharge',
+};
+
+// Since the billing category is passed into a RateCalculator as part of the
+// arguments, this enum exists for documentation rather than actual type safety.
+// Once this compiles to JS, nothing will prevent an application from passing in
+// an abitrary string for the billingCategory.
+export enum BillingCategory {
+  TAX = 'tax',
+  SUPPLY = 'supply',
+  DELIVERY = 'delivery',
+}
```

Comment:
this might not even be worth enumerating... in practice, it will be defined at runtime so it doesn't really give us any type safety.